src/NetTopologySuite.Lab/Hull/ChiShape.cs

```
@@ -0,0 +1,151 @@
+                CheckMaxEdgeAndListBorderTriangles(quadEdgeTriangleList, triangle);
+            }
+
+            while (quadEdgeTriangleList.Count != 0)
+            {
+                var triangle = quadEdgeTriangleList.Last();
+                var edge = triangle.Key;
+                Debug.WriteLine(triangle.Value);
+                quadEdgeTriangleList.Remove(edge);
+                if (edge.Length.CompareTo(_tolerance) >= 0 && RemovalGivesRegularPolygon(triangle.Value))
+                {
+                    var neighbours = triangle.Value.GetNeighbours();
+                    triangle.Value.Kill();
+                    foreach (var neighbour in from n in neighbours
+                                              where n != null && n.IsLive()
+                                              select n)
+                    {
+                        Debug.WriteLine(neighbour);
+                        CheckMaxEdgeAndListBorderTriangles(quadEdgeTriangleList, neighbour);
+                    }
+                }
+            }
+            var lineMerger = new LineMerger();
+            var fact = new GeometryFactory();
+            foreach (var triangle in tris)
+            {
+                if (!triangle.IsLive()) continue;
+                for (int i = 0; i < 3; i++)
+                {
+                    if (triangle.GetAdjacentTriangleAcrossEdge(i) == null || !triangle.GetAdjacentTriangleAcrossEdge(i).IsLive())
+                    {
+                        lineMerger.Add(triangle.GetEdge(i).ToLineSegment().ToGeometry(fact));
+                    }
+                }
+            }
+            var mergedLine = (LineString)lineMerger.GetMergedLineStrings().FirstOrDefault();
+            if (mergedLine.IsRing)
+            {
+                var lr = new LinearRing(mergedLine.CoordinateSequence, fact);
+
+                var concaveHull = new Polygon(lr, null, fact);
+
+                return concaveHull;
+            }
+            return mergedLine;
```

Documentation says that this is supposed to throw if we can't generate a hull, but this returns a 1-dimensional geometry... which is the correct behavior?

---

Let's just throw an exception and call it a day.